### DISCUSSION TITLE
Is TTGO T-Call ESP32 SIM800L supported ?

### DISCUSSION THREAD

**Discussion by trex2000** [arendst/Tasmota#15168] [Ideas]
Hello,
I've searched over the internet, but could not find an answer. Is the following board supported by tasmota ?
http://www.lilygo.cn/claprod_view.aspx?TypeId=62&Id=1403&FId=t28:62:28
https://www.aliexpress.com/item/33045221960.html

What I want to achieve is to build an ibeacon that sends nearby tags via MQTT to Home Assistant. For internet connection, I would like to use a SIM card instead of Wifi connection.

**Comment by sfromis** [comment]
If you want Tasmota to do the usual wifi communication via the mobile radio instead, not supported.

To use the feature at all, my expectation is that it is AT style serial communication, and with Tasmota (not done it myself) you can generally bridge simple serial communication to JSON payloads, or handle it in a Berry script.

**Reply by trex2000** [reply]
I see, I was trying to do something like this:
https://randomnerdtutorials.com/esp32-cloud-mqtt-broker-sim800l/

**Reply by barbudor** [reply]
This TinyGSM library seems interresting.
It looks like it makes the wideband network part of the network stack of the ESP so PubSubClient can use it the same as Wifi or Ethernet.
I know that Espressif SDK supports some wideband features and includes some examples about using a SIM800L, but haven't looked in details.
It might be possible to implement a driver for that.

**Comment by Schrotti1963** [comment]
I am also very interested on working SIM800l in Tasmota. Is it possible to implement a driver, so that I can make a MQTT-connection over GSM instead over WIFI?

**Reply by barbudor** [reply]
I started to look at this but put it aside for now
It mainly depends if you want the GSM/Broadband only for MQTT or for everything
Remenber that Tasmota needs some network features to work properly such as NTP. So it you have no Wifi at all it might get complicated.
For me ESP8266 is almost out of the picture. It is not that TinyGSM wouldn't run on ESP8266 (it runs on ATMega so no problems) but it would be complicated to integrate it into Tasmota.

The alternative would be with ESP32 as the Espressif SDK can support PPP to get a proper complete network connection over GSM. Just like you can have Ethernet instead of WIfi on ESP32, you could have Broadband network in a complete transparent way for Tasmota.
However there are some problems:
- It is not included by default in the Arduino platform for ESP32 and you need a custom build (I started to get help from Jason on that but for now I've put that aside as I didn't had real need)
- If your Tasmota is fully exposed to the Broadband network, there is a security issue as Tasmota do not include any firewall

So far I've decided to buy a cheap Wifi/Broadband router for 30€ which provide full 4G access for not much more than a SIM800 module.
Less headaches

**Reply by Schrotti1963** [reply]
Thanks for the fast answere. 
My intention is to built a cheep GPS tracker that provides positiondata over GSM with mqtt.
At the moment a NEO-6M works on a ESP8266 very good. WIFI is provided by a raspberry pi zero wh and a SIM800L as GSM/WIFI-Router.

**Reply by barbudor** [reply]
Are you using Tasmota on the ESP8266 ?
I wouldn't have considered Tasmota for a tracker by why not ?

If you have a GPS, then you don't need NTP access so a driver with integration of TinyGSM could be possible... Need to check how much of specific code would be required

There are other use cases for single sensor/actuator with no default wifi access. While a SIM800 adapter could be attractive, I'm not sure it worth vs simple WIfi/GSM router stick.

I Purchased this European version without USB adapter for less than 25€ : https://www.aliexpress.com/item/1005003123615228.html

**Comment by Romik4x4** [comment]
I am also very interested in the support of GSM in Tasmot for emergency situations. For example, a power outage or breakdown of network equipment. Perhaps simple functions will suffice to start with.

**Reply by barbudor** [reply]
A basic driver in Berry could send AT commands for SMS or http requests

**Reply by sfromis** [reply]
Yeah, it would be quite doable for someone with the hardware and interest to create a basic driver in Berry, implementing commands to send, with the usual methods of making issuing commands. Of course, also publish payloads if/when something is received, available to subscribers and rules. Berry does make serial communication simple to integrate.

**Comment by vinibali** [comment]
Here is another thread, which talks about the AT commands: https://github.com/arendst/Tasmota/issues/5170

**Reply by pkkrusty** [reply]
The documentation from SimCom for the 800L has a full list of AT commands, including a sequence which should open TCP/UDP connections to IP addresses. I'm going to experiment with sending one-way GPS data through a TCP link to node-red as a start. Assuming most carriers put mobile connections behind CGNAT, seems like the insecure nature of Tasmota wouldn't be a huge deal if the main devs want to hook in GPRS network. However, I think these $3 800L modules only do 2G GPRS data which means it's going the way of the dodo. Good enough for MQTT and sensor data...

**Comment by pkkrusty** [comment]
I keep waiting for someone smarter than me to integrate this into the Tasmota network stack, or put together a simple set of commands to send serial data. But since 2G is gone in Europe and North America, the main devs don't have much incentive. 

Anyone have some examples in Berry or scripting they want to share?

**Reply by vinibali** [reply]
The big operators in Hungary are still providing 2G radio here at least, IIRC some public services are using GPRS for communication.

**Comment by Jason2866** [comment]
With this PR https://github.com/espressif/arduino-esp32/pull/9518 support for PPP modems has been added to Arduino core 3.0.0. Building a Tasmota Arduino framework including this driver was successful.
Could establish an Internet connection via an simcom 7600 LTE Module. Cheaper GSM Sim800 modules are supported too. Since the Arduino driver is designed as another network driver it is not to complicated to integrate to use with Tasmota. To harden Tasmota to be secure is the part where the work is waiting.

**Reply by sfromis** [reply]
Beware of being that guy..... 
![image](https://github.com/arendst/Tasmota/assets/47082390/1593b370-c698-4da7-82b7-eee387ae8b1b)

**Reply by pkkrusty** [reply]
Good news! Exposed to the internet, turning off webserver and just using MQTT for commands/data probably a safer option. I look forward to enjoying the driver someone eventually writes while I contribute nothing.

**Reply by s-hadinger** [reply]
It may be useful to disable ping response as well. Although I don't think there is an option currently available for this

**Reply by ATP-IOT** [reply]
Hello. I'm very interested on the support for PPP modems. How I can test this? Any download for the firmware or anything? How to add this to the firmware manually or something?

**Reply by sfromis** [reply]
Seems that this has not been fulfilled....
> I look forward to enjoying the driver someone **eventually** writes
